solver.rb

```diff
@@ -0,0 +1,31 @@
+class Solver
+  def factorial(num)
+    facto = 1
+    if num.class != Integer || num.negative?
+      'exception'
+    elsif num <= 1
+      1
+    else
+      (2..num).each do |i|
+        facto *= i
+      end
+      facto
+    end
+  end
+
+  def reverse(string)
+    string.reverse
+  end
+
+  def fizzbuzz(num)
+    if (num % 3).zero? && (num % 5).zero?
+      'fizzbuzz'
+    elsif (num % 3).zero?
+      'fizz'
+    elsif (num % 5).zero?
```

Comment:
- [ ] Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD. currently your commit history show otherwise, the idea is to write tests first and then the code. This will be verified in your commit history, so it is important to use the proper sequence. Kindly fix this [Check project requirement here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=Remember%20to%20start%20with%20tests%20for%20all%20of%20the%20methods%20described%20above.%20Your%20commit%20history%20should%20prove%20that%20you%20have%20used%20TDD.) 👍🏽